codec/encode.go

```diff
@@ -539,6 +544,23 @@ func encStructFieldKey(ee encDriver, keyType valueType, s string) {
 	}
 }
 
+// A structField stores all the info for a single field of a struct.
+// If encoding into a map, name is filled, and otherwise it is left
+// empty. For known fields, val is filled, and for unknown fields
+// (used only when encoding into a map), encodedVal is filled.
+type structField struct {
```

Comment:
This is replacing `stringRV`, right?  Should it go in `helper.go` in its place?  I'm not sure but I don't think there are any other uses of `stringRV` left, are there?

Reply:
`stringRv` is still used for maps with string keys